content/docs/ckan/harvester/fair-data-point-harvester-update-strategy/_index.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Fair Data Point Harvester Update Strategy
+---
+
+<!--
+SPDX-FileCopyrightText: 2024 Stichting Health-RI
+
+SPDX-License-Identifier: CC-BY-4.0
+-->
+
+{{< toc-tree >}}
+
+* * *
+
+As was [mentioned](../../harvester/_index.md), a CKAN harvester must implement `gather_stage`, `fetch_stage` and `import_stage`.
+
+During `gather_stage` the fair data point harvester requests all the available resources from a source. 
+For each resource a unique guid is generated. Guids are generated by the harvester as the following:
+- for a catalog `catalog=<FDP link to a catalog>`
+- for a dataset `catalog=<FDP link to the dataset's parent catalog>;dataset=<FDP link to a dataset>`
+
+where `FDP link to a catalog/dataset` is an FDP reference URL (subject URL) of the resource. 
+e.g. for the following dataset https://health-ri.sandbox.semlab-leiden.nl/dataset/d7129d28-b72a-437f-8db0-4f0258dd3c25 
+CKAN harvester guid will be `catalog=https://health-ri.sandbox.semlab-leiden.nl/catalog/e3faf7ad-050c-475f-8ce4-da7e2faa5cd0;dataset=https://health-ri.sandbox.semlab-leiden.nl/dataset/d7129d28-b72a-437f-8db0-4f0258dd3c25`.
```

Comment:
Why is `dcterm:identification` not used here for dataset identification instead of this FDP identification? Using `dcterm:identification` would keep datasets unique and avoid issues when a dataset is moved from one catalog (like ONS) to another. This assumes that `dcterm:identification` is unique. Can you explain the rationale behind this choice?

it is not in the scope of a spike to change it of course 

What I miss in de the document when the dataset/catalog is updated. After pressing the button re-harvest? Could you clarify ?  

Further clear document how CRUD operations work

Reply:
I can not say why exactly the described strategy was chosen (bc I wasn't the one who implemented that) but I can guess that uniqueness of `dcterm:identifier` is not guaranteed. It up to the source which uploads data to FDP, it is not checked and not managed by FDP and even was not mandatory. An FDP link is unique, however, combining dataset id with catalog id is to me a redundant overengineering.

It is in the document: stages are the same - get two lists on gather, select intersection, assign status "change" and call action update_package.

Reply:
Ok, clear!

Regarding the update, there was some strange behavior with the re-harvest button. Could you describe what happened? I understand that behind the scenes, update_package is called. Could you clarify the process? Or am I missing something?

Reply:
Ah, got it. To me it should be in another section (regarding cron job set up).